### DISCUSSION TITLE
Vim freezes using mini surround while recording a Macro

### DISCUSSION THREAD

**Discussion by ViniciusLagoGehrke** [LazyVim/LazyVim#4887] [Q&A]
The first question would be how could I debug this since I'm not sure where the issue happens.

- I can use the mini.surround without a problem.
- I can record macros without a problem.
- As soon as I press `gs` while recording a macro vim freezes and I have to terminate the Wezterm/Tmux pane.

**Comment by dpetka2001** [comment]
Cannot reproduce. Screencast following

https://github.com/user-attachments/assets/21d5c301-4c92-4a13-9a75-80ed95455e34

Also maybe a question better asked in `mini.nvim` repo if it's actually a bug, since it won't be fixed here.

**Comment by pkarlovsek** [comment]
I have a similar problem:

The first question would also be how to debug this since I'm unsure where the issue happens.

- I can use the mini.surround without a problem.
- I can record macros without a problem.
- When I press `g` while recording a macro, which-key is not showing commands as opposed to if I don't record the macro. But the main problem is that when I am recording a macro after pressing `gsa` nvim acts as I would press only `a` i.e. goes to append mode. The expected behavior would be that I could specify range e.g. `iw` and then mini.surround asks for surround character.

I found a workaround. While recording a macro I first do a visual selection, then `:lua MiniSurrount.add('visual')<CR>`, then I type selecting surrounding character e.g. `"`, quit recording macro. Now I have a working macro.

**Reply by alexandre-abrioux** [reply]
I had the same issue, the fix for me was to type `gsa` faster, like described here: https://github.com/echasnovski/mini.nvim/issues/1385#issuecomment-2551363094

**Reply by Buttars** [reply]
This is still an issue. Pressing the binds quickly is a workaround but which-key should definitely be showing the options.
